prototypes/brownian_01.ipynb

```diff
@@ -0,0 +1,925 @@
+       "  (epoch=2022-01-09 00:00:00.000 x=1.935475453149731 km y=11.828073927563336 km),\n",
+       "  (epoch=2022-01-10 00:00:00.000 x=5.222701961898467 km y=0.868684884635686 km),\n",
+       "  (epoch=2022-01-11 00:00:00.000 x=4.823137392858772 km y=4.440253533694463 km),\n",
+       " ]>)"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "c = b.propagate(TimeDelta(7 << u.d))\n",
+    "b, c"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "57334aab-2032-4abb-8c49-4faafbd8caba",
+   "metadata": {},
+   "source": [
+    "Because a single \"scalar\" `Orbit` object can be propagated to many points in time, we can now actually test it:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "76ab6a32-a6ae-48a3-9001-6cc9e9ef50a8",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<Orbit epoch=2022-01-01 00:00:00.000 x=2.0 km y=3.0 km>"
+      ]
+     },
+     "execution_count": 18,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "a = Orbit(State(Time('2022-01-01 00:00', scale = 'ut1'), 2.0 << u.km, 3.0 << u.km))\n",
+    "a"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "id": "f3f4d7b6-093c-4bc6-bd75-047308cfa738",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<OrbitArray shape=(2, 3) value=[\n",
+       " (epoch=2022-01-02 00:00:00.000 x=1.8758431019286572 km y=2.876872646005249 km),\n",
+       " (epoch=2022-01-03 00:00:00.000 x=2.4438721928780947 km y=3.169198727393347 km),\n",
+       " (epoch=2022-01-04 00:00:00.000 x=2.361379575267373 km y=3.3804708701261568 km),\n",
+       " (epoch=2022-01-05 00:00:00.000 x=0.9873495799265113 km y=4.303186379469707 km),\n",
+       " (epoch=2022-01-06 00:00:00.000 x=0.8682743746407637 km y=3.2713790948420023 km),\n",
+       " (epoch=2022-01-07 00:00:00.000 x=4.8200590299603645 km y=4.356403526953536 km),\n",
+       "]>"
+      ]
+     },
+     "execution_count": 19,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "b = a.propagate(TimeDelta(np.arange(1, 7).reshape(2, 3) << u.d))\n",
+    "b"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "beb5b6bd-199a-45ba-8b68-e81201f68948",
+   "metadata": {},
+   "source": [
+    "## Further thinking\n",
+    "\n",
+    "- [Structured arrays](https://numpy.org/doc/stable/user/basics.rec.html) can actually be n-dimensional. This could be a foundation for `StateArray` classes, opening the back door for alternative libraries which are exposing an [array interface](https://numpy.org/doc/stable/reference/arrays.interface.html). In a bigger picture, the state array could be the place where stuff like `cupy.ndarray` or `dask.array` are transparently accepted.\n",
```

Comment:
This is very interesting idea - are structured arrays something that has a future in the NumPy roadmap? Maybe we could rethink some things on top of this.

However, epochs are going to be an issue.

Reply:
NumPy datatime is known to have lots of issues, or that's what I recall from reading Twitter threads and GitHub comments. I don't have any pointers at hand.

Reply:
Notice that time of flight is _not_ used by the propagation algorithm, despite the ill-designed poliastro `core` functions:

https://github.com/poliastro/poliastro/blob/726c95d17da3fb6b43bf38ace5b2b17170e1255f/src/poliastro/core/propagation/farnocchia.py#L290-L298

Astropy should take care of adding times, not us. This deserves opening an issue.

Reply:
Deleted a highly misleading comment I made: the time of flight is _definitely_ used by the propagation algorithm.

Reply:
To @s-m-e's point: by passing `tof` (elapsed time in seconds) to the propagation algorithms, we have already chosen a unit. Therefore, one could take this approach one step further and work with elapsed seconds everywhere internally, to avoid the alleged performance issues of `astropy.time` and the difficulties to store them.

I _guess_ this approach would work as long as folks don't want to propagate trans-Neptunian objects or the Solar System barycenter with respect to the Milky Way. But, if it doesn't work, it's already slightly wrong in the current implementation.

Important reading: https://docs.astropy.org/en/stable/time/index.html#internal-representation

> The Time object maintains an internal representation of time as a pair of double precision numbers expressing Julian days. The sum of the two numbers is the Julian Date for that time relative to the given time scale. 

And about what would happen by using a single-float representation:

> Users requiring no better than microsecond precision over human time scales (~100 years) can safely ignore the internal representation details and skip this section.

On the other hand, about time scales:

> We have shown in the above that the difference between two UTC times is a TimeDelta with a scale of TAI. This is because a UTC time difference cannot be uniquely defined unless the user knows the two times that were differenced (because of leap seconds, a day does not always have 86400 seconds). For all other time scales, the TimeDelta inherits the scale of the first Time object.

Therefore,

1. There is a way to _do it right_ (in other words, aim for maximum accuracy) with relatively simple tools, but it requires some care.
2. We might (I haven't deeply analyzed it) have tiny precision bugs in the code at the moment, given that we are not being particularly careful with the `TimeDelta` scales and just performing a `to(u.s)` to pass the value to the propagator:

https://github.com/poliastro/poliastro/blob/726c95d17da3fb6b43bf38ace5b2b17170e1255f/src/poliastro/twobody/orbit.py#L1098-L1099

https://github.com/poliastro/poliastro/blob/726c95d17da3fb6b43bf38ace5b2b17170e1255f/src/poliastro/twobody/propagation.py#L516-L520

On the other hand, we might be getting too deep into this topic, and I don't want this to block our efforts.

Reply:
> The Time object maintains an internal representation of time as a pair of double precision numbers expressing Julian days. The sum of the two numbers is the Julian Date for that time relative to the given time scale.

Then I was not digging deep enough. Interesting. This would actually allow a really clean implementation.